django/contrib/staticfiles/management/commands/collectstatic.py

```diff
@@ -218,7 +218,15 @@ def clear_dir(self, path):
                          smart_text(fpath), level=1)
             else:
                 self.log("Deleting '%s'" % smart_text(fpath), level=1)
-                self.storage.delete(fpath)
+                full_path = self.storage.path(fpath)
+                if not os.path.exists(full_path) and os.path.lexists(full_path):
+                    # Delete if the file is a broken symlink (see #25346)
```

Comment:
The ticket reference isn't necessary (we only include that if it's a really obscure issue that can't be easily explained in comments).